pkg/scheduler/scheduler_test.go

```diff
@@ -415,6 +416,40 @@ func TestFailureHandler_PodAlreadyBound(t *testing.T) {
 	}
 }
 
+// TestWithPercentaeOfNodesToScore to test set global PercentagrOfNodesToScore to KubeSchedulerConfig
+func TestWithPercentageOfNodesToScore(t *testing.T) {
+	tests := []struct {
+		name                           string
+		percentageOfNodesToScore       *int32
+		option                         *schedulerOptions
```

Comment:
create this object inside the loop

Reply:
How?

Reply:
Actually, forget about this suggestion.
Can you instead call `New(..., WithPercentageOfNodesToScore)` and then verify that the scheduler is created with the appropriate value?

`schedulerOptions` is an internal object, so it's not much worth testing at that level.

Reply:
Rewrote the test function to create a scheduler using `WithPercentageOfNodesToScore`.